resource_activity/models/resource_activity.py

```diff
@@ -678,11 +677,18 @@ def prepare_lines(self, activity):
 
     def _create_sale_order(self, activity, partner_id):
         SaleOrder = self.env['sale.order']
+        sale_note = SaleOrder._default_note()
+        for line in activity.location_id.line_ids:
+            if (line.note_id
+                    and line.location_id == activity.location_id
+                    and line.activity_type_id == activity.activity_type):
+                sale_note = line.note_id.content
         order_id = SaleOrder.create({

```

Comment:
From the sql constraint, I see there is one line at most matching this condition. For this, I would have done a `filtered` instead of a loop.

Reply:
Either that or you could also make a computed field (store=False) `sale_note` that depends on `activity.location_id` and `activity.activity_type`.

But this way is fine.

Reply:
if you set a computed field depending on other fields it must be True.

Reply:
I'll try `filtered` if time allows it

Reply:
`filtered` done with a additional check on the `active` field